exclause/with.go

```diff
@@ -45,12 +46,12 @@ func (with With) Build(builder clause.Builder) {
 		if index > 0 {
 			builder.WriteByte(',')
 		}
-		cte.Build(builder)
+		cte.Build(builder, with.Materialized)
 	}
 }
 
 // Build build CTE
-func (cte CTE) Build(builder clause.Builder) {
+func (cte CTE) Build(builder clause.Builder, materialized bool) {
```

Comment:
If possible, I would prefer not to change the public interface. When the materialized property
moves to the CTE struct, the Build method won't need the materialized argument.